torchgeo/datasets/cyclone.py

```diff
@@ -125,10 +125,12 @@ def __getitem__(self, index: int) -> dict[str, Any]:
         )
 
         sample: dict[str, Any] = {'image': self._load_image(directory)}
-        sample.update(self._load_features(directory))
+        features = self._load_features(directory)
+        sample['label'] = features['label']
 
         if self.transforms is not None:
             sample = self.transforms(sample)
+            sample.update({x: features[x] for x in features if x != 'label'})
```

Comment:
Assumes that transforms will only be applied on kornia keys i.e. 'image', 'mask', 'class', 'boxes' etc.

Reply:
Would be nice if kornia's `AugmentationSequential` ignores keys are that are not relevant to it while accepting a dict input.

Reply:
Not opposed to supporting both, but maybe save that for a follow-up PR if that isn't required by this PR.

Reply:
I don't recall us specifying datasets only support torchvision-style transforms and thought we were moving towards kornia. 

This was done because when a batch is passed through our kornia augmentations in `on_batch_after_transfer` it can only have kornia compatible keys.

Reply:
With kornia/kornia#2971, we should be able to pass keys and have kornia only act on the kornia compatible keys and ignore the rest.